exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -142,25 +141,25 @@ public ParquetGroupScan(String userName,
       metadataProviderManager = new FileSystemMetadataProviderManager();
     }
 
-    ParquetFileTableMetadataProviderBuilder builder =
-        (ParquetFileTableMetadataProviderBuilder) metadataProviderManager.builder(MetadataProviderManager.MetadataProviderKind.PARQUET_TABLE);
+    ParquetTableMetadataProviderBuilder<?> builder =
+        tableMetadataProviderBuilder(metadataProviderManager);
 
-    this.metadataProvider = builder
+    ParquetTableMetadataProvider metadataProvider = builder
         .withSelection(selection)
         .withReaderConfig(readerConfig)
         .withFileSystem(fs)
         .withCorrectCorruptedDates(formatConfig.areCorruptDatesAutoCorrected())
         .build();
 
-    ParquetTableMetadataProvider parquetTableMetadataProvider = (ParquetTableMetadataProvider) this.metadataProvider;
-    this.usedMetadataCache = parquetTableMetadataProvider.isUsedMetadataCache();
-    this.usedMetastore = parquetTableMetadataProvider instanceof MetastoreParquetTableMetadataProvider;
-    this.selectionRoot = parquetTableMetadataProvider.getSelectionRoot();
-    this.entries = parquetTableMetadataProvider.getEntries();
-    this.fileSet = parquetTableMetadataProvider.getFileSet();
+    this.usedMetadataCache = metadataProvider.isUsedMetadataCache();
+    this.usedMetastore = metadataProviderManager.usesMetastore();
+    this.selectionRoot = metadataProvider.getSelectionRoot();
+    this.entries = metadataProvider.getEntries();
+    this.fileSet = metadataProvider.getFileSet();
+    this.metadataProvider = metadataProvider;
```

Comment:
These are the same six attributes mentioned for the generic Easy group scan. Rather than duplicating them, can we encapsulate them in a single `MetadataResult` class? (You pick the name.)